cromwell-drs-localizer/src/main/scala/drs/localizer/downloaders/BulkAccessUrlDownloader.scala

```diff
@@ -0,0 +1,138 @@
+package drs.localizer.downloaders
+
+import cats.effect.{ExitCode, IO}
+import cloud.nio.impl.drs.{AccessUrl, DrsResolverResponse}
+import com.typesafe.scalalogging.StrictLogging
+
+import java.nio.charset.StandardCharsets
+import java.nio.file.{Files, Path, Paths}
+import scala.sys.process.{Process, ProcessLogger}
+import scala.util.matching.Regex
+import drs.localizer.ResolvedDrsUrl
+case class GetmResult(returnCode: Int, stderr: String)
+/**
+  * Getm is a python tool that is used to download resolved DRS uris quickly and in parallel.
+  * This class builds a getm-manifest.json file that it uses for input, and builds/executes a shell command
+  * to invoke the Getm tool, which is expected to already be installed in the local environment.
+  * @param resolvedUrls
+  */
+case class BulkAccessUrlDownloader(resolvedUrls : List[ResolvedDrsUrl]) extends Downloader with StrictLogging {
+  /**
+    * Write a json manifest to disk that looks like:
+    * // [
+    * //  {
+    * //     "url" : "www.123.com",
+    * //     "filepath" : "path/to/where/123/should/be/downloaded",
+    * //     "checksum" : "sdfjndsfjkfsdjsdfkjsdf",
+    * //     "checksum-algorithm" : "md5"
+    * //  },
+    * //  {
+    * //     "url" : "www.567.com"
+    * //     "filepath" : "path/to/where/567/should/be/downloaded",
+    * //     "checksum" : "asdasdasfsdfsdfasdsdfasd",
+    * //     "checksum-algorithm" : "md5"
+    * //  }
+    * // ]
+    *
+    * @param resolvedUrls
+    * @return Filepath of a getm-manifest.json that Getm can use to download multiple files in parallel.
+    */
+  def generateJsonManifest(resolvedUrls : List[ResolvedDrsUrl]): IO[Path] = {
+    def toJsonString(drsResponse: DrsResolverResponse, destinationFilepath: String): String = {
+      //NB: trailing comma is being removed in generateJsonManifest
+      val accessUrl: AccessUrl = drsResponse.accessUrl.getOrElse(AccessUrl("missing", None))
+      drsResponse.hashes.map(_ => {
+        val checksum = GetmChecksum(drsResponse.hashes, accessUrl).value.getOrElse("error_calculating_checksum")
+        val checksumAlgorithm = GetmChecksum(drsResponse.hashes, accessUrl).getmAlgorithm
+        s"""  {
```

Comment:
Not necessarily a blocker for this PR, but we should look into letting a library take care of the JSON serialization here. Doing it "by hand" is going to be a lot more brittle.

Reply:
I agree!